### PR TITLE
chore: Replacing private Documentation URLs with public ones

### PR DESCRIPTION
**Pull request by utmishra**
The Documentation links in `guideline-block-settings/src/index.ts` are private URLs (like `https://developer.frontify.com/document/1366/.....` ) and won't be accessible to the general public. Changed them with the public ones.

### CODE REVIEW

File: packages/guideline-blocks-settings/src/index.ts [hunk 1]
```diff
@@ -27,6 +26,7 @@ import type {
     TextareaBlock as TextareaBlockSidebarSettings,
     ValueOrPromisedValue as ValueOrPromisedValueSidebarSettings,
 } from '@frontify/sidebar-settings';
+import type { FC } from 'react';
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/index.ts:29]: Please revert the change and set your IDE to not change these lines, it is noise and really not useful.

File: packages/guideline-blocks-settings/src/index.ts [hunk 2]
```diff
@@ -75,34 +75,34 @@ export type BlockSettingsStructureExport = {
 export type BlockProps = {
     /**
      * The Frontify App Bridge provides an interface to the Frontify app internals.
-     * {@link https://developer.frontify.com/document/1366#/details-concepts/content-blocks/introducing-the-appbridge}
+     * {@link https://developer.frontify.com/d/XFPCrGNrXQQM/content-blocks#/details-concepts-1/content-blocks/introducing-the-app-bridge}
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/index.ts:78]: Could you check to fix the slug `detail-concepts-1` to be cleaner like `detail-concepts` ? 🤔

Comment (utmishra) [reply packages/guideline-blocks-settings/src/index.ts:78]: This might change later anyway.
According to @floriangaechter:
> We're planning on moving the developer platform to a dedicated tool anyway sometime this year and that would involve a new path-scheme (especially in regards to the coming additions to the brand sdk)
